### PR TITLE
refactor(ir): replace enum functionality with string

### LINKED ISSUE #4477: refactor(ir): replace enum functionality with string
It is being used only in the impala codebase.

### PR DESCRIPTION
**Pull request by cpcloud**
This PR aliases Enum to String. See commit messages for details. Closes #4477.

### CODE REVIEW

**Comment by kszucs** [comment]
@cpcloud the impala backend has failing tests

**Comment by cpcloud** [comment]
Turns out that `dt.Category` is providing some value in the form of tracking the cardinality of expressions.

For example, bucketing an expression using a list of bins of say, length 4, will be tracked in the cardinality property of the `Category` dtype which allows validation of the labels to the `CategoryValue.label` API.

I'm going to leave Category as-is for now and just do the enum aliasing in this PR.

File: ibis/expr/datatypes/core.py [hunk 1]
```diff
@@ -995,4 +986,5 @@ class INET(String):
     macaddr=macaddr,
     inet=inet,
     decimal=decimal,
+    Enum=Enum,
```
Comment (kszucs) [inline ibis/expr/datatypes/core.py:989]: `Enum=String`? Shouldn't we deprecate it?

**Comment by kszucs** [comment]
> I'm going to leave Category as-is for now and just do the enum aliasing in this PR.

Need to update the Pr's title then

**Comment by cpcloud** [comment]
@kszucs Done, but looks like there's still some failing impala tests. Looking 👀

**Review by kszucs — APPROVED** [review]

**Pull request merged**